targets/TARGET_Realtek/TARGET_AMEBA/RTWInterface.cpp

```diff
@@ -75,7 +75,7 @@ static rtw_result_t scan_result_handler( rtw_scan_handler_result_t* malloced_sca
             ap.channel = record->channel;
             WiFiAccessPoint *accesspoint = new WiFiAccessPoint(ap);
             memcpy(&scan_handler->ap_details[scan_handler->ap_num], accesspoint, sizeof(WiFiAccessPoint));
-            delete[] accesspoint;
+            delete accesspoint;
         }
         scan_handler->ap_num++;
```

Comment:
@Archcady Placement new does not allocate memory from the heap. It use the address supplied in the placement param to construct the object. 

In this specific case, the object will be constructed at the address `&scan_handler->ap_details[scan_handler->ap_num]`. You'll have to call the delete operator when the `ap_details` memory block goes away.